src/Parsing/ParserState.php

```diff
@@ -220,18 +220,26 @@ public function parseCharacter($bIsForIdentifier)
     }
 
     /**
-     * @return array<int, Comment>|void
+     * Consumes whitespace and comments and returns the comments.
+     * If $withoutComment is true, only whitespace is consumed.
+     *
+     * @param bool $withoutComment Do not consume comments, only whitespace.
+     *
+     * @return array<int, Comment>|void List of comments.
      *
      * @throws UnexpectedEOFException
      * @throws UnexpectedTokenException
      */
-    public function consumeWhiteSpace(): array
+    public function consumeWhiteSpace(bool $withoutComment = false): array
```

Comment:
I'm not a big fan of boolean function parameters that switch behavior. Instead, please let's have a separate, well-named method, e.g., `comsumeWhitespaceWithoutComment` or something alike.

Reply:
Maybe we can then also move the new method to `Rule`. (I'm not 100% sure about this and would welcome your thoughts on this, @ziegenberg @sabberworm @JakeQZ.)

Reply:
I would go even further and rename the current function `comsumeWhitespace` to `comsumeWhitespaceWithComment` and create a new function `comsumeWhitespace` which precisely only consumes white space. I was baffled to see `comsumeWhitespace` also consume comments and return them.

Reply:
Yes, this would probably be a breaking change. So, I introduced the boolean function parameter to stay within the current wording (which I deem wrong) and still make the change non-breaking.

I would consider having `comsumeWhitespace` and a new `comsumeWhitespaceWithoutComment` a not-so-good solution. That would cement the semantically wrong situation.

Reply:
I like the idea of renaming this method!

Would you be willing to do this as a (or more specific), two pre-patch PRs?

1. one (to be backported) that renames the method, adds the specific type annotation, and re-adds the old method to redirect to the new method, and marks it as `@deprecated will be removed in version 9.0` (including a corresponding line in the changelog
2. another (not to be backported) that removes the old method (including a corresponding entry in the changelog)

I'd be willing to review these two pre-PRs quickly so we can rebase on them.

We also might want to mark the renamed method as `@internal` as it needs to be `public` for technical reasons, but is not expected to be called from outside this library. What do you think?

Reply:
So you want `comsumeWhitespaceWithComment` and `comsumeWhitespaceWithoutComment`?

Reply:
Yes, exactly!

Reply:
I am not a big fan of `comsumeWhitespaceWithoutComment`, having a function name telling you what it does not do. But I'll create the two pre-patch PRs as asked.

Reply:
Ah, it seems I misread your comment and got confused. :exploding_head: 

Okay, so I'm fine with this plan:

> I would go even further and rename the current function `comsumeWhitespace` to `comsumeWhitespaceWithComment` and create a new function `comsumeWhitespace` which precisely only consumes white space.

We'll need to do this in a non-breaking way for the parts that we backport, though.

Reply:
> All `consume*` functions are in the class `ParserState`, so why move it to `Rule`?

I've thought about this some more and now think having it in `ParserState` is perfectly fine. :-)